src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,443 @@
+            }
+
+            if (VSTestTestAdapterPath != null && VSTestTestAdapterPath.Length > 0)
+            {
+                foreach (var arg in VSTestTestAdapterPath)
+                {
+                    allArgs.Add("--testAdapterPath:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+                }
+            }
+
+            if (!string.IsNullOrEmpty(VSTestFramework))
+            {
+                allArgs.Add("--framework:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestFramework));
+            }
+
+            // vstest.console only support x86 and x64 for argument platform
+            if (!string.IsNullOrEmpty(VSTestPlatform) && !VSTestPlatform.Contains("AnyCPU"))
+            {
+                allArgs.Add("--platform:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestPlatform));
+            }
+
+            if (!string.IsNullOrEmpty(VSTestTestCaseFilter))
+            {
+                allArgs.Add("--testCaseFilter:" +
+                            ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestTestCaseFilter));
+            }
+
+            if (VSTestLogger != null && VSTestLogger.Length > 0)
+            {
+                foreach (var arg in VSTestLogger)
+                {
+                    allArgs.Add("--logger:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+
+                    if (arg.StartsWith("console", StringComparison.OrdinalIgnoreCase))
+                    {
+                        isConsoleLoggerSpecifiedByUser = true;
+                    }
+                }
+            }
+
+            if (!string.IsNullOrEmpty(VSTestResultsDirectory))
+            {
+                allArgs.Add("--resultsDirectory:" +
+                            ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestResultsDirectory));
+            }
+
+            if (!string.IsNullOrEmpty(VSTestListTests))
+            {
+                allArgs.Add("--listTests");
+            }
+
+            if (!string.IsNullOrEmpty(VSTestDiag))
+            {
+                allArgs.Add("--Diag:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestDiag));
+            }
+
+            if (string.IsNullOrEmpty(TestFileFullPath))
+            {
+                Log.LogError("Test file path cannot be empty or null.");
+            }
+            else
+            {
+                allArgs.Add(ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(TestFileFullPath));
+            }
+
+            // Console logger was not specified by user, but verbosity was, hence add default console logger with verbosity as specified
+            if (!string.IsNullOrEmpty(VSTestVerbosity) && !isConsoleLoggerSpecifiedByUser)
+            {
+                var normalTestLogging = new List<string>() { "n", "normal", "d", "detailed", "diag", "diagnostic" };
```

Comment:
If you're only doing `Contains` operations, these should be `HashSet<string>` instead.

Also, these should be made `static` to avoid allocating each time the task runs